### PR TITLE
[DPO] Refactor eval logging of dpo trainer

### PR DESCRIPTION
**Pull request by mnoukhov**
Address part 1 of #952 by removing `store_metrics` and switching to more standard `compute_metrics` letting `Trainer` handle logging

### CODE REVIEW

**Comment by lvwerra** [comment]
Thanks for working on this! Just ping me whenever you would like to run the CI.

**Comment by mnoukhov** [comment]
@lvwerra, I completed this part of the refactor, can you run the CI?
 
@kashif as part of this, I removed logging all the `logits` since it seemed a bit excessive to pass around the `batch x seq_len x vocab size` array and taking the `mean` didn't feel like a meaningful thing to log since we already have `logps`. Let me know if you'd like this reversed

**Comment by kashif** [comment]
@mnoukhov are you also logging sample outputs as well here?

**Comment by mnoukhov** [comment]
@kashif I didn't change sample output logging in this PR, so it's still there. I was going to address the sample logging issue (2 in #952) in a second PR to keep things managable

**Comment by kashif** [comment]
so this PR LGTM! feed free to merge

**Comment by kashif** [comment]
@mnoukhov so folks really want the additional metrics to be logged during training which isn't possible at the moment, so lets revert this for now and continue this refactoring to figure that out.

**Comment by mnoukhov** [comment]
I don't think this change prevents additional metrics from being logged. It feels like it actually makes it easier to log

**Comment by kashif** [comment]
Ok so let's try to add the previous behavior if possible... I couldn't figure it out at first glance... I think the trainer's training loop doesn't seem to have a way to do it... can you kindly confirm?

**Comment by mnoukhov** [comment]
Basically what #1046 wants is to log metrics during training, which is not how `Trainer` usually works but it is a common ask e.g. https://discuss.huggingface.co/t/logging-training-accuracy-using-trainer-class/5524/3. The reverted code does not do this correctly.

AFAIK the problem with trying to put `self.log` into `compute_loss` is it doesn't really make sense whenever you use multi-gpu or gradient accumulation. In both cases you're calling `compute_loss` more than once per step and the previous code just logged the metrics for one subset of that batch (e.g. the last minibatch in gradient accumulation or one gpu's batch). @lewtun, is this right?

If you want metrics for the training set, I would recommend doing `evaluate` on the training set any time it is also done on the test set, using a callback.
